cli/lib/tasks/download.js

```diff
@@ -9,23 +9,14 @@ const request = require('@cypress/request')
 const Promise = require('bluebird')
 const requestProgress = require('request-progress')
 const { stripIndent } = require('common-tags')
+const getProxyFromURI = require('./getProxyFromURI')
```

Comment:
Can you update the implementation to use the `proxy-from-env` node module instead of inlining `getProxyFromURI`? We use this library elsewhere so might as well keep using it here: https://github.com/Rob--W/proxy-from-env

Reply:
Sure, I've adjusted the implementation.

I've wrapped the method from that module to also be able to use the npm configured proxies. It behaves a bit different now though (see tests):
- http_proxy and HTTP_PROXY env are only used on http requests
- https_proxy and HTTPS_PROXY env are only used on https requests
- npm_config_https_proxy and npm_config_proxy are used if no other proxy was selected by the `proxy-from-env` module. https is preferred as before
- npm_config_https_proxy and npm_config_proxy are used with both http and https requests if set
- npm_config_https_proxy and npm_config_proxy do NOT respect `NO_PROXY`